contract/src/lib.rs

```diff
@@ -418,69 +516,46 @@ impl MpcContract {
         }
     }
 
-    /// This is the root public key combined from all the public keys of the participants.
-    pub fn public_key(&self) -> PublicKey {
-        match &self.protocol_state {
-            ProtocolContractState::Running(state) => state.public_key.clone(),
-            ProtocolContractState::Resharing(state) => state.public_key.clone(),
-            _ => env::panic_str("public key not available (protocol is not running or resharing)"),
-        }
-    }
-
-    pub fn version(&self) -> String {
-        env!("CARGO_PKG_VERSION").to_string()
-    }
-
-    /// Key versions refer new versions of the root key that we may choose to generate on cohort changes
-    /// Older key versions will always work but newer key versions were never held by older signers
-    /// Newer key versions may also add new security features, like only existing within a secure enclave
-    /// Currently only 0 is a valid key version
-    pub const fn latest_key_version(&self) -> u32 {
-        0
-    }
-
-    fn add_signature(&mut self, payload: &[u8; 32], signature: (String, String)) {
-        if self.pending_requests.contains_key(payload) {
-            self.pending_requests.insert(payload, &Some(signature));
-        }
-    }
-
-    fn add_request(&mut self, payload: &[u8; 32], signature: Option<(String, String)>) {
-        if self.request_counter > 8 {
-            env::panic_str("Too many pending requests. Please, try again later.");
-        }
-        if !self.pending_requests.contains_key(payload) {
-            self.request_counter += 1;
-        }
-        self.pending_requests.insert(payload, &signature);
-    }
-
-    fn remove_request(&mut self, payload: &[u8; 32]) {
-        self.pending_requests.remove(payload);
-        self.request_counter -= 1;
-    }
-
-    // Helper functions
     #[private]
     #[init(ignore_state)]
     pub fn clean(keys: Vec<near_sdk::json_types::Base64VecU8>) -> Self {
         log!("clean: keys={:?}", keys);
         for key in keys.iter() {
             env::storage_remove(&key.0);
         }
-        Self {
+        Self::V0(MpcContract {
             protocol_state: ProtocolContractState::NotInitialized,
             pending_requests: LookupMap::new(b"m"),
             request_counter: 0,
-        }
+        })
     }
 
     #[private]
     pub fn clean_payloads(&mut self, payloads: Vec<[u8; 32]>, counter: u32) {
-        log!("clean_payloads");
-        for payload in payloads.iter() {
-            self.pending_requests.remove(payload);
+        match self {
+            Self::V0(mpc_contract) => {
+                mpc_contract.clean_payloads(payloads, counter);
+            }
         }
-        self.request_counter = counter;
+    }
+
+    /// This is the root public key combined from all the public keys of the participants.
+    pub fn public_key(self) -> PublicKey {
```

Comment:
We should proxy all user-facing functions. All the tmp and service functions are not that important.

Reply:
what do you mean by proxy all user-facing functions?

Reply:
We care about breaking changes for user and node-facing functions. That is why we should wrap all of them. All the service functions can have breaking changes.